<launch>
	<!-- global rosparam -->
	<param name="use_sim_time" value="true"/>

	<!-- rosbag -->
	<node pkg="rosbag" type="play" name="player" args="--clock /home/amsl/ozaki/bagfiles/sq2/2018-12-11-13-37-56.bag"/>

	<!-- rviz -->
	<!-- <node pkg="rviz" type="rviz" name="rviz_startpoint" args="&#45;d /home/amsl/.rviz/trajectories_startpoint.rviz"/> -->
	<!-- <node pkg="rviz" type="rviz" name="rviz_skyview" args="&#45;d /home/amsl/.rviz/trajectories_skyview.rviz"/> -->

	<!-- sq_lidar -->
	<node pkg="sq_lidar" name="sq_lidar" type="sq_lidar">
		<param name="fixed_frame" value="/base_link"/>
		<remap from="~/scan" to="/sq_lidar/scan"/>
		<remap from="~/cloud" to="/cloud"/>
	</node>
	<node pkg="sq_lidar_pc_store" name="sq_lidar_pc_store" type="sq_lidar_pc_store">
		<remap from="~/odom" to="/combined_odometry"/>
	</node>

	<!-- proposed method -->
	<node pkg="pose_estimation_walls" name="imu_initial_alignment" type="imu_initial_alignment" output="screen"/>
	<node pkg="pose_estimation_walls" name="ekf_pose" type="ekf_pose" output="screen"/>
	<node pkg="pose_estimation_walls" name="pose_estimation_gaussian_spheres" type="pose_estimation_gaussian_spheres" output="screen">
		<remap from="/velodyne_points" to="/sq_lidar/stored"/>
	</node>
	<node pkg="pose_estimation_walls" name="combine_linearvelocity_and_pose" type="combine_linearvelocity_and_pose"/>

	<!-- comparisons	 -->
	<node pkg="pose_estimation_walls" name="gyrodometry" type="gyrodometry"/>
	<!-- <node pkg="pose_estimation_walls" name="odom_creater_for_lsdslam" type="odom_creater_for_lsdslam"/> -->

	<!-- graph -->
	<!-- <node pkg="pose_estimation_walls" name="graph_rpy" type="graph_rpy.py"/> -->
</launch>
